src/ultk/language/grammar.py

```diff
@@ -59,6 +52,7 @@ def __str__(self) -> str:
         return out_str
 
 
+@dataclass(eq=True, kw_only=True)
```

Comment:
Is it confirmed that `eq=True` behaves as desired with this inheritance? I.e. that equality will also check for `form` and `meaning` which are defined only in `Expression` but not here?  Mainly a question about how dataclasses work :)